### PR TITLE
Fixing some older procs not respecting resistance flags.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Makes machinery and structures `take_damage()` respect `resistance_flags`, purging the older breakable and destroyable vars. And fixes things being able to fly through prison hull windows/shutters I guess.

## Why It's Good For The Game
Fixes a couple bugs, namely around runners and tanks.

## Changelog
:cl:
fix: Fixes a couple old procs not respecting the INDESTRUCTIBLE and UNACIDABLE resistance flags.
fix Fixes things going through prison hull windows frames and shutters.
/:cl:

### CODE REVIEW

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -603,10 +602,7 @@ obj/machinery/proc/med_scan(mob/living/carbon/human/H, dat, var/list/known_impla
 
 //Damage
 /obj/machinery/proc/take_damage(dam)
-	if(!destructible)
-		return
-
-	if(!dam)
+	if(!dam || CHECK_MULTIPLE_BITFIELDS(resistance_flags, INDESTRUCTIBLE|UNACIDABLE))
```
Comment (Menshin) [inline code/game/machinery/machinery.dm]: That means that the `machinery`  has to be **both** `INDESTRUCTIBLE` and `UNACIDABLE` for this to pass.
Intended ?

File: code/game/objects/structures.dm [hunk 2]
```diff
@@ -32,16 +31,17 @@
...
+	if(CHECK_BITFIELD(resistance_flags, INDESTRUCTIBLE))
+		return
+	if(HULK in user.mutations)
+		user.say(pick(";RAAAAAAAARGH!", ";HNNNNNNNNNGGGGGGH!", ";GWAAAAAAAARRRHHH!", "NNNNNNNNGGGGGGGGHH!", ";AAAAAAARRRGH!" ))
+		visible_message("<span class='danger'>[user] smashes the [src] apart!</span>")
+		destroy_structure()
 
 /obj/structure/attackby(obj/item/C as obj, mob/user as mob)
 	. = ..()
-	if(istype(C, /obj/item/tool/pickaxe/plasmacutter) && !user.action_busy && breakable && !CHECK_BITFIELD(resistance_flags, UNACIDABLE|INDESTRUCTIBLE))
```
Comment (spookydonut) [inline code/game/objects/structures.dm]: change this to early return

File: code/game/objects/structures.dm [hunk 3]
```diff
@@ -55,13 +55,13 @@
 	return FALSE
 
 /obj/structure/attack_animal(mob/living/user)
-	if(breakable)
-		if(user.wall_smash)
-			visible_message("<span class='danger'>[user] smashes [src] apart!</span>")
-			destroy_structure()
+	. = ..()
+	if(CHECK_BITFIELD(resistance_flags, INDESTRUCTIBLE) && user.wall_smash)
```
Comment (spookydonut) [inline code/game/objects/structures.dm]: logic on this is inverted, also early return

File: code/game/objects/structures/tables_racks.dm [hunk 4]
```diff
@@ -255,22 +254,23 @@
...
-		if(health <= 0)
-			M.visible_message("<span class='danger'>\The [M] slices [src] apart!</span>", \
-			"<span class='danger'>You slice [src] apart!</span>", null, 5)
-			destroy_structure()
-		else
-			M.visible_message("<span class='danger'>[M] slashes [src]!</span>", \
-			"<span class='danger'>You slash [src]!</span>", null, 5)
-		if(M.stealth_router(HANDLE_STEALTH_CHECK)) //Cancel stealth if we have it due to aggro.
-			M.stealth_router(HANDLE_STEALTH_CODE_CANCEL)
+	if(!CHECK_BITFIELD(resistance_flags, INDESTRUCTIBLE))
```
Comment (spookydonut) [inline code/game/objects/structures/tables_racks.dm]:
```suggestion
	if(CHECK_BITFIELD(resistance_flags, INDESTRUCTIBLE))
```

**Pull request merged**